### ISSUE TITLE
[Sepolia] 0% on SNAP Phase 1 for a long time because of bad peers

### ISSUE THREAD

**Issue by kamilchodola** [NethermindEth/nethermind#5014]
**Describe the bug**
During a sync procedure Sepolia on SNAP sync was always lightning fast (like in minute or two).
Now I've noticed that it takes long time to even start Snap Phase 1 - it remains at 0% for very long time.
Attaching logs from this

**Desktop (please complete the following information):**
 - OS: Ubuntu
 - Chain: Sepolia
 - Version: Master

**Logs**
[sepolia.logs.zip](https://github.com/NethermindEth/nethermind/files/10231585/sepolia.logs.zip)

**Comment by asdacap** [comment]
They seems to be stucked on a peer. and keep timing out. Unit test shows that the lesser peer check *should* sleep that particular peer and it should try other peer. It is also possible that the snap sync timed out as there is a block download happening at the same time. Blocks are large and because of how the networking currently works, until the blocks is deserialized, other request (snap) is blocked.

**Comment by asdacap** [comment]
No.. sepolia is small so the block is also small... something else is causing the timeout.

**Comment by kamilchodola** [comment]
@asdacap 
Unfortunately this still happens on 1.16.0-unstable which includes this fix so there must be other root cause for this.
Here are logs from Nethermind<--->Teku
[SepoliaPeersIssue.zip](https://github.com/NethermindEth/nethermind/files/10368319/SepoliaPeersIssue.zip)

**Comment by kamilchodola** [comment]
@asdacap 
It eventually unstucked after almost an hour, here are some more logs from this node if You want to analyze:
[SepoliaSnapSyncUnstucked.zip](https://github.com/NethermindEth/nethermind/files/10368431/SepoliaSnapSyncUnstucked.zip)
